src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,236 @@
+        match.addColorMatch(Constants.Conveyor.NONE);
+        position.add(DriverStation.Alliance.Invalid.name());
+        position.add(DriverStation.Alliance.Invalid.name());
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * @return the color sensor value as a DriverStation.Alliance enum.
+     */
+    public DriverStation.Alliance getColor() {
+        Color color = colorSensorIntake.getColor();
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+        SmartDashboard.putBoolean("match", resultColor == Color.kRed);
+
+        if (resultColor == Constants.Conveyor.RED) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Constants.Conveyor.BLUE) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;
+        }
+    }
+
+    /**
+     * @return the amount of balls inside the cargo.
+     */
+    public int getCargoCount() {
+        return position.size();
+    }
+
+    /**
+     * Set motor power.
+     *
+     * @param power the power to the motor [-1, 1].
+     */
+    public void setPower(double power) {
+        motor.set(power);
+    }
+
+    /**
+     * open the "flap" - solenoid
+     */
+    public void openFlap() {
+//        flap.set(false);
+    }
+
+    /**
+     * closes the "flap" - solenoid
+     */
+    public void closeFlap() {
+//        flap.set(true);
+    }
+
+    /**
+     * toggles the "flap" - solenoid
+     */
+    public void toggle() {
+//        flap.toggle();
+    }
+
+    private void updateActualBallPositions() throws InterruptedException {
+        var colorIntake = getColor();
+        boolean currentPostFlapBeamInput = postFlapBeam.get();
+        SmartDashboard.putString("alliance", colorIntake.name());
+        double power = 1;
+//        double power = motor.getMotorOutputPercent();
+        if (currentPostFlapBeamInput && !lastPostFlapBeamInput && power > 0) {
+            if(!position.toArray()[0].equals(DriverStation.Alliance.Invalid.name())) {
+                position.poll();
+                var temp = position.take();
+                position.add(DriverStation.Alliance.Invalid.name());
+                position.add(temp);
+            } else {
+                position.poll();
+                position.poll();
+                position.add(DriverStation.Alliance.Invalid.name());
+                position.add(DriverStation.Alliance.Invalid.name());
+            }
+        }
+        if (colorIntake != lastSeenColor && !colorIntake.equals(DriverStation.Alliance.Invalid)) {
+            if (power < 0) {
+                if (position.remainingCapacity() == 0) {
+                    position.poll();

```

Comment:
If the queue is empty, why do you need to poll?

Reply:
The queue isn't empty. See comment above.